### PR TITLE
Make rotation state save when saving schematic coordinates

### LINKED ISSUE #14557: Schematica Save Coordinates doesn't save rotation or flip
### Your GTNH Discord Username

electrobot

### Your Pack Version

2.4.0

### Your Server

Stonelegion

### Java Version

Java 8

### Type of Server

Crucible

### Your Expectation

Checked the "Save Coordinates" feature added to Schematica and it doesn't save rotation and flip.

### The Reality

The "Save Coordinates" feature added to Schematica only saves the server name, schematic name, x/y/z co-ords. 

### Your Proposal

Add flip and rotation to the save function.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [X] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

### PR DESCRIPTION
**Pull request by Lyfts**
This simply makes it so the rotation of your schematic is saved upon saving coordinates.
I've tested this for a while now and think all the kinks have been sorted.

Schematics I used for testing:
[schematics.zip](https://github.com/GTNewHorizons/Schematica/files/12704602/schematics.zip)

The test one can be used in dev env and contains both TE and blocks, it's easy to observe rotation due to the skulls.
The EHE one is a behemoth and has to be in full pack. It was mostly used to see if large schematics worked fine.

Closes: https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/14557

Flip is mentioned in the ticket and likewise has a button in the GUI but is not actually implemented in 1.7.10.

### CODE REVIEW

File: src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java [hunk 1]
```diff
@@ -206,8 +206,8 @@ public static boolean saveCoordinatesFile(Map<String, Map<String, Map<String, In
         }
     }
 
-    public static boolean addCoordinates(String worldServerName, String schematicName, Integer X, Integer Y,
-            Integer Z) {
+    public static boolean addCoordinates(String worldServerName, String schematicName, Integer X, Integer Y, Integer Z,
```
Comment (mist475) [inline src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java]: As this now also saves the rotation it should be reflected in the method name, "addCoordinatesAndRotation" should work well enough

File: src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java [hunk 2]
```diff
@@ -247,7 +249,7 @@ public static boolean addCoordinates(String worldServerName, String schematicNam
      * @return {@link ImmutablePair} with bool (true if coordinates found, false if not) and {@link ImmutableTriple}
```
Comment (mist475) [inline src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java]: This is now outdated as you replaced it with a Triple

**Review by mist475 — CHANGES_REQUESTED** [review]
Thank you for this, this is something I should've thought of when I added coordinate saving to this fork. I was initially worried this would break with schematics that had previously been saved but I can confirm it works as expected.

You could edit the message printed so it mentions it also saved the rotation but I'll leave that up to you to decide

File: src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java (hunk 2 above)
Comment (Lyfts) [reply src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java]: I'm a dummy at docs :) 
I have pushed a change, feel free to let me know if it needs further changes.

**Review by mist475 — APPROVED** [review]

File: src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java [hunk 3]
```diff
@@ -257,19 +264,20 @@ public static ImmutablePair<Boolean, ImmutableTriple<Integer, Integer, Integer>>
                     Map<String, Integer> coordinateMap = schematicMap.get(schematicName);
                     if (coordinateMap.containsKey("X") && coordinateMap.containsKey("Y")
                             && coordinateMap.containsKey("Z")) {
-                        return new ImmutablePair<>(
+                        return new ImmutableTriple<>(
                                 true,
+                                coordinateMap.get("Rotation"),
```
Comment (eigenraven) [inline src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java]: Will this work if you load an old file that didn't have the Rotation field?

Comment (mist475) [reply src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java]: It does work in my testing, though it does print an error in the logs which I missed when I reviewed, please replace with
```java
.getOrDefault("Rotation",0)
```

Comment (Lyfts) [reply src/main/java/com/github/lunatrius/schematica/proxy/ClientProxy.java]: Don't know how i didn't notice that. 
Thanks for spotting it :)

**Review by eigenraven — APPROVED** [review]

**Pull request merged**